src/mame/apple/apple2.cpp

```diff
@@ -119,15 +119,24 @@ class apple2_state : public driver_device
 	virtual void machine_start() override;
 	virtual void machine_reset() override;
 
-	template <bool Invert, bool Flip>
-	u32 screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
-	u32 screen_update_jp(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
-	u32 screen_update_ultr(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
-
-	u32 screen_update_ff(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect) { return screen_update<false, false>(screen, bitmap, cliprect); }
-	u32 screen_update_ft(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect) { return screen_update<false, true>(screen, bitmap, cliprect); }
-	u32 screen_update_tf(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect) { return screen_update<true, false>(screen, bitmap, cliprect); }
-	u32 screen_update_tt(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect) { return screen_update<true, true>(screen, bitmap, cliprect); }
+	u32 screen_update_ff(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect) {
+		return m_video->screen_update<a2_video_device::model::II, false, false>(screen, bitmap, cliprect);
+	}
+	u32 screen_update_ft(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect) {
+		return m_video->screen_update<a2_video_device::model::II, false, true>(screen, bitmap, cliprect);
+	}
+	u32 screen_update_tf(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect) {
+		return m_video->screen_update<a2_video_device::model::II, true, false>(screen, bitmap, cliprect);
+	}
+	u32 screen_update_tt(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect) {
+		return m_video->screen_update<a2_video_device::model::II, true, true>(screen, bitmap, cliprect);
+	}
+	u32 screen_update_jp(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect) {
+		return m_video->screen_update<a2_video_device::model::II_J_PLUS, true, true>(screen, bitmap, cliprect);
+	}
+	u32 screen_update_ultr(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect) {
+		return m_video->screen_update<a2_video_device::model::IVEL_ULTRA, true, false>(screen, bitmap, cliprect);
+	}
```

Comment:
I had to parenthesize the argument of `FUNC` because the macro system doesn't understand the commas. As a result the quoted function name will be parenthesized. Does that matter?

Reply:
`FUNC` doesn't work with more than one template parameter, so the quoted function name will be `(&cls::method<args>)` instead of `cls::method<args>`. Does that matter? I copied this hack from sun/sun3.cpp which is the only other instance of this problem I could find.

Reply:
Using `NAME((&...))` is the preferred way to deal with multiple template arguments and the preprocessor – you’ll find lots of instances of it when creating tilemaps where the tile info callback function has multiple template parameters.